src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt

```diff
@@ -268,11 +269,7 @@ open class Komga(suffix: String = "") : ConfigurableSource, UnmeteredSource, Htt
             title = metadata.title
             url = "$baseUrl/api/v1/series/$id"
             thumbnail_url = "$url/thumbnail"
-            status = when (metadata.status) {
-                "ONGOING" -> SManga.ONGOING
-                "ENDED" -> SManga.COMPLETED
-                else -> SManga.UNKNOWN
-            }
+            status = parseStatus(metadata, booksCount)
```

Comment:
```suggestion
            status = when {
                metadata.status == "ENDED" && metadata.totalBookCount != null && booksCount < metadata.totalBookCount -> SManga.PUBLISHING_FINISHED
                metadata.status == "ENDED" -> SManga.COMPLETED
                metadata.status == "ONGOING" -> SManga.ONGOING
                metadata.status == "ABANDONED" -> SManga.CANCELLED
                metadata.status == "HIATUS" -> SManga.ON_HIATUS
                else -> SManga.UNKNOWN
            }
```

I think this would be simpler to understand. In your code below, the reader must mentally go through temp states to understand what's going on, IMHO it's much more difficult to read.

Reply:
Better use `when (metadata.status)` and in the `"ENDED" ->` branch, use another `when` or `if (condition) A else B`.

Reply:
Nested when is kinda ugly